infiniband-diags/ibdiag_common.c

```diff
@@ -670,10 +672,15 @@ static int resolve_guid(char *ca_name, uint8_t ca_port, ib_portid_t *portid,
 	if (resolve_self(ca_name, ca_port, NULL, NULL, &selfgid) < 0)
 		return -1;
 
-	memcpy(&prefix, portid->gid, sizeof(prefix));
-	if (!prefix)
-		mad_set_field64(portid->gid, 0, IB_GID_PREFIX_F,
-				IB_DEFAULT_SUBN_PREFIX);
+	memcpy(&prefix, selfgid, sizeof(prefix));
+	gid_prefix = htobe64(prefix);
+	prefix = IB_DEFAULT_SUBN_PREFIX;
+	if (gid_prefix)
+		p_gid_prefix = &gid_prefix;
+	else
+		p_gid_prefix = &prefix;
+	mad_encode_field(portid->gid, IB_GID_PREFIX_F,
```

Comment:
What? Why is this like this? It should be:

```
__be64 prefix;

memcpy(&prefix, portid->gid, sizeof(prefix));
mad_set_field64(portid->gid, 0, IB_GID_PREFIX_F,
￼				prefix:be64toh(prefix)?IB_DEFAULT_SUBN_PREFIX);
```

Reply:
Fixed